Foundation/include/Poco/ProcessRunner.h

```diff
@@ -0,0 +1,168 @@
+//
+// ProcessRunner.h
+//
+// Library: Foundation
+// Package: Processes
+// Module:  ProcessRunner
+//
+// Definition of the ProcessRunner class.
+//
+// Copyright (c) 2023, Applied Informatics Software Engineering GmbH.
+// Aleph ONE Software Engineering d.o.o.,
+// and Contributors.
+//
+// SPDX-License-Identifier:    BSL-1.0
+//
+
+
+#ifndef Foundation_ProcessRunner_INCLUDED
+#define Foundation_ProcessRunner_INCLUDED
+
+
+#include "Poco/Foundation.h"
+#include "Poco/Process.h"
+#include "Poco/ProcessOptions.h"
+#include "Poco/Runnable.h"
+#include "Poco/Thread.h"
+#include "Poco/Logger.h"
+#include "Poco/Stopwatch.h"
+#include <atomic>
+#include <vector>
+
+
+namespace Poco {
+
+
+class Foundation_API ProcessRunner: public Poco::Runnable
+	/// ProcessRunner is a wrapper class for `Poco::ProcessHandle.
+	/// It starts and terminates a process with enabled or disabled (default)
+	/// `stdio` pipes, and waits for process PID to be created before
```

Comment:
change to: and **optionally** waits for process PID to be created before

Reply:
Fixed in [ca79ba8](https://github.com/pocoproject/poco/pull/4225/commits/ca79ba8cb058dba942dd9b86e912037325339d6e)